src/displayapp/screens/settings/SettingDisplay.cpp

```diff
@@ -43,10 +43,10 @@ SettingDisplay::SettingDisplay(Pinetime::Applications::DisplayApp* app, Pinetime
   lv_label_set_align(icon, LV_LABEL_ALIGN_CENTER);
   lv_obj_align(icon, title, LV_ALIGN_OUT_LEFT_MID, -10, 0);
 
-  char buffer[12];
+  char buffer[4];
   for (unsigned int i = 0; i < options.size(); i++) {
     cbOption[i] = lv_checkbox_create(container1, nullptr);
-    sprintf(buffer, "%2ds", options[i] / 1000);
+    snprintf(buffer, sizeof(buffer), "%2" PRIu8 "s", options[i] / 1000);
```

Comment:
I've just tried running these changes on my watch (without using `snprintf`), and these changes break the settings menu. The checkboxes are stacked with only 1 column, rather than 2, and the text on all of them is `hus`. I don't currently have too much time to look in to what's happening here.

Reply:
To fix this, the format specifier macro used here needs to be changed from `PRIu8` to `PRIu16`.

Reply:
Yes, indeed. I think the reason that it was made to be u8 was because any u16 / 1000 can be fit into a u8.

Reply:
This is done to keep the character buffer small, I think?

Then if we want the `PRIu8` the result of `u16/1000` could be wrapped by a `static_cast<uint8_t>()`

Reply:
I changed it to `PRIu16`. `char buffer[4]` saves buffer space and `snprintf` ensures security. Unnecessary to use `PRIu8`.